modules/tensor_mechanics/src/materials/DiscreteRRIsotropicPlasticity.C

```diff
@@ -0,0 +1,150 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "DiscreteRRIsotropicPlasticity.h"
+
+template<>
+InputParameters validParams<DiscreteRRIsotropicPlasticity>()
+{
+  InputParameters params = validParams<DiscreteRadialReturnStressIncrement>();
+
+  // Linear strain hardening parameters
+  params.addParam<Real>("yield_stress", "The point at which plastic strain begins accumulating");
+  params.addParam<FunctionName>("yield_stress_function", "Yield stress as a function of temperature");
+  params.addParam<Real>("hardening_constant", "Hardening slope");
+  params.addParam<FunctionName>("hardening_function", "True stress as a function of plastic strain");
+  params.addCoupledVar("temp", "Coupled Temperature");
+
+  return params;
+}
+
+
+DiscreteRRIsotropicPlasticity::DiscreteRRIsotropicPlasticity( const InputParameters & parameters)
+  :DiscreteRadialReturnStressIncrement(parameters),
+   _yield_stress_function(isParamValid("yield_stress_function") ? &getFunction("yield_stress_function") : NULL),
+   _yield_stress(isParamValid("yield_stress") ? getParam<Real>("yield_stress") : 0),
+   _hardening_constant(isParamValid("hardening_constant") ? getParam<Real>("hardening_constant") : 0),
```

Comment:
You'll find this particular method useful:
http://mooseframework.org/docs/doxygen/moose/classInputParameters.html#a8cf3c6dcf652bdf074d835012f27254f

On Thu, Mar 17, 2016 at 11:23 AM Stephanie Pitts <notifications@github.com>
wrote:

> In modules/tensor_mechanics/src/materials/DiscreteRRIsotropicPlasticity.C
> <https://github.com/idaholab/moose/pull/6602#discussion_r56543164>:
>
> > +  // Linear strain hardening parameters
> > +  params.addParam<Real>("yield_stress", "The point at which plastic strain begins accumulating");
> > +  params.addParam<FunctionName>("yield_stress_function", "Yield stress as a function of temperature");
> > +  params.addParam<Real>("hardening_constant", "Hardening slope");
> > +  params.addParam<FunctionName>("hardening_function", "True stress as a function of plastic strain");
> > +  params.addCoupledVar("temp", "Coupled Temperature");
> > +
> > +  return params;
> > +}
> > +
> > +
> > +DiscreteRRIsotropicPlasticity::DiscreteRRIsotropicPlasticity( const InputParameters & parameters)
> > +  :DiscreteRadialReturnStressIncrement(parameters),
> > +   _yield_stress_function(isParamValid("yield_stress_function") ? &getFunction("yield_stress_function") : NULL),
> > +   _yield_stress(isParamValid("yield_stress") ? getParam<Real>("yield_stress") : 0),
> > +   _hardening_constant(isParamValid("hardening_constant") ? getParam<Real>("hardening_constant") : 0),
>
> In this code we have options for the user to define a yield stress or a
> yield stress function but not both and a hardening function or a hardening
> constant but not both--the error messages make allowable combinations more
> clear. With defining a default value, the parameters will always be valid,
> and the error messages won't trigger. Is there another way that you would
> recommend to write the error messages so that I can use the default value
> and complete the error checks?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/6602/files/5e2f395c45979d61f75e72845fa277986429ab2c#r56543164>
>